doc/MEMORY_CONSTRAINED_CONFIGURATIONS.md

```diff
@@ -0,0 +1,31 @@
+# Memory Constrained Environments
+
+In certain situations you may be running TemporalX in a memory constrained environments. When using default configurations, TemporalX is already supremely memory efficient, however there are certain situations where you may want to scale up this memory efficiency, say on a raspberry pi. Although efficient on memory, the configurations in this guide will provide a somewhat noticeable impact to performance, and as such is only recommended in memory constrained environments.
```

Comment:
I think we should describe the memory requirements for constrained and none constrained configurations first, as this is the most useful information.

Reply:
good idea